### ISSUE TITLE
Question: Might the pre-training of fisher lead to a source-free setting instead of a test-time setting?

### ISSUE THREAD

**Issue by CM-BF** [mr-eggplant/EATA#3]
Dear EATA authors,

Thank you for your great paper and code! I really enjoy your fascinating work.

I'm here to ask a question about your code. I noticed that the pre-training of fisher is done on the test set, which means the method accesses test data twice (one for the fisher pre-training, one for the test-time adaptation). Therefore, with the fisher pre-training, we may regard EATA as a source-free adaptation method instead of a test-time adaptation method. Could you let me know whether I'm correct or not? Could you tell me the specific reason why we need to pre-train the fisher on the test set? Is it possible for us to train the fisher on other sets (e.g., ImageNet) and obtain similar results?

I appreciate you patient and any answers!

Best,
Shurui

**Comment by mr-eggplant** [comment]
Hi Shurui,

Thank you for your interest!

We collect a small set of in-distribution test samples (i.e., the clean validation set of ImageNet ) to calculate Fisher. Analyses regarding the sample number needed can be found in Figure 4, where we found that 500 samples are enough. 

Once Fisher is calculated, we keep it fixed and apply it to any type of distribution shift (e.g., the whole ImageNet-C, consisting of 15 corruption types and 5 severity levels). Therefore, I think EATA does not need to see OOD test samples twice, and it belongs to the test-time setting, rather than source-free.

Also, we can also use the training set of ImageNet to calculate Fisher if the data are available. This would achieve similar performance.

Kindly let me know if you have further questions!

Best,
Shuaicheng
